### ISSUE TITLE
Apple Music: cannot play, errors "Failed to load because no supported source was found"

### ISSUE THREAD

**Issue by alexboutros** [jiahaog/nativefier#989]
**Bug description**
Nativefied Apple Music and it loads up as expected, but when I try to play music I get the error "Failed to load because no supported source was found".

- [x] I looked at `nativefier --help` and https://github.com/jiahaog/nativefier/blob/master/docs/api.md
- [x] I searched existing open & closed issues. Yes, my bug is new.
- [x] I'm using the latest version available at https://github.com/jiahaog/nativefier/releases

**Steps to reproduce**
1. nativefier -n "Apple Music" https://music.apple.com
2. ./AppleMusic
3. Play any song

**Expected behavior**
Music should play

**Actual behavior**
Error "Failed to load because no supported source was found". Music will not play.

**Context**
 - Nativefier 8.0.7
 - Node.js v10.20.1
 - Npm v 6.14.5
 - OS: Manjaro Linux

**Comment by BananaManCJ** [comment]
Apple Music uses DRM-protected music which won't play with a nativefier app. I recommend using a Chrome/Firefox app from WebCatalog instead.

**Comment by TinyBabyOwl** [comment]
> 
> 
> Apple Music uses DRM-protected music which won't play with a nativefier app. I recommend using a Chrome/Firefox app from WebCatalog instead.

I had a worry it was something like that. Thanks for the tip on WebCatalog that looks super handy.

**Comment by alexboutros** [comment]
> 
> 
> Apple Music uses DRM-protected music which won't play with a nativefier app. I recommend using a Chrome/Firefox app from WebCatalog instead.

I had a worry it was something like that. Thanks for the tip on WebCatalog that looks super handy.

**Comment by BananaManCJ** [comment]
> I had a worry it was something like that. Thanks for the tip on WebCatalog that looks super handy.

I'd still recommend using nativefier for other apps that don't use DRM protection because WebCatalog requires you to pay if you want to have more than 2 apps and nativefier has more options.

**Comment by alexboutros** [comment]
> > I had a worry it was something like that. Thanks for the tip on WebCatalog that looks super handy.
> 
> I'd still recommend using nativefier for other apps that don't use DRM protection because WebCatalog requires you to pay if you want to have more than 2 apps and nativefier has more options.

Yeah I’ll still use nativefier for everything else. Webcatalog isn’t really working out for me anyways. When I try to use the chrome option it tells me chrome isn’t installed (it is) and same for brave. The Firefox option just launches a full Firefox window, tabs and all. 

I’ll just have to use Apple Music from the browser for the time.

**Comment by sarweshparajuli** [comment]
You need to add `--honest` to end of the code while compiling. e.g. `nativefier --name "Music" "https://music.apple.com" --honest`.

Then, it worked for me.

**Comment by flying-sausages** [comment]
Fairly certain that if the electron is modified to use whatever WebCatalog is using, DRM playback should be doable.

**Comment by khulaifi95** [comment]
I've the same issue and I found that it is because the chromium is missing Widevine cdm library. You can check for it once you download an app called 'apple-music-for-linux' on snap store. It performs bad but it enlightened me that it will install this cdm library on first run. I think tweaking the chromium used in electron could fix it.

**Comment by GitGangGuy** [comment]
I'd be really happy if somebody who knows the nativefier codebase could add drm support. The only thing needed would be a flag which installs the widevine module with the electron app (though I'm not sure if it's easy wiring it together, I've never done anything in electron before :/ ). Just came here after trying to get a native-feel apple music client on linux.

**Comment by flying-sausages** [comment]
I suppose this is tied up in the Discussion in #440. Widevine needs to be added. I'm trying to decipher what blocked the PR but can't find it.

**Comment by GitGangGuy** [comment]
According to the electron docs, the user could also pass a path to the widevine library file (downloaded to a specific Firefox/Chromium/Spyware directory after having enabled DRM playback) which can then be used to play DRM restricted content within an electron app.

**Comment by BananaManCJ** [comment]
> According to the electron docs, the user could also pass a path to the widevine library file (downloaded to a specific Firefox/Chromium/Spyware directory after having enabled DRM playback) which can then be used to play DRM restricted content within an electron app.

Yeah, I grabbed the Widevine library from Google Chrome so I could play DRM content in Ungoogled Chromium, and since electron uses the chromium codebase that could work.

**Comment by Skipper1931** [comment]
Sorry to ressurect this, but I did some snooping and apparently they added a "--widevine" that fixed the issue (cant use Apple Music) for me.

**Comment by flying-sausages** [comment]
Not sure why i just got a notification for this but `--widevine` is now a supported flag in nativefier. This can be closed i assume.

**Comment by ronjouch** [comment]
> Not sure why i just got a notification for this but `--widevine` is now a supported flag in nativefier. This can be closed i assume.

@flying-sausages true, thanks! Closing, see https://github.com/jiahaog/nativefier/blob/master/docs/api.md#widevine . If that doesn't work, do shout and I'll re-open the issue.

**Comment by Neo11Neo** [comment]
I also tried converting apple music but it did not work with passing the widevine argument. The natified version with widewine would not open any window and when I tried closing it with task manager, it restarted. Without widewine the app would work normally until you try to play music. Same issue. I am on Nativefier Version 46.1.1. I also checked for widewine being up to date and located in the right chrome directory.

Looking forward to some of your ideas.
